src/netaddress.cpp

```diff
@@ -494,76 +494,46 @@ uint64_t CNetAddr::GetHash() const
     return nRet;
 }
 
-// private extensions to enum Network, only returned by GetExtNetwork,
-// and only used in GetReachabilityFrom
-static const int NET_UNKNOWN = NET_MAX + 0;
-static const int NET_TEREDO  = NET_MAX + 1;
-int static GetExtNetwork(const CNetAddr *addr)
-{
-    if (addr == nullptr)
-        return NET_UNKNOWN;
-    if (addr->IsRFC4380())
-        return NET_TEREDO;
-    return addr->GetNetwork();
-}
+// private extension to enum Network, only returned by GetExtNetwork, and only
+// used in GetReachabilityFrom
+#define NET_UNKNOWN NET_MAX
 
 /** Calculates a metric for how reachable (*this) is from a given partner */
 int CNetAddr::GetReachabilityFrom(const CNetAddr *paddrPartner) const
 {
     enum Reachability {
         REACH_UNREACHABLE,
         REACH_DEFAULT,
-        REACH_TEREDO,
-        REACH_IPV6_WEAK,
         REACH_IPV4,
-        REACH_IPV6_STRONG,
+        REACH_IPV6,
         REACH_PRIVATE
     };
 
     if (!IsRoutable() || IsInternal())
         return REACH_UNREACHABLE;
 
-    int ourNet = GetExtNetwork(this);
-    int theirNet = GetExtNetwork(paddrPartner);
-    bool fTunnel = IsRFC3964() || IsRFC6052() || IsRFC6145();
+    int ourNet = GetNetwork();
```

Comment:
If ourNet is `nullptr` you will now mark its reachability to NET_IPV6 which may be the highest-scored one and announce this one (but can we have null entries in mapLocalHost anyway ?)

Reply:
Perhaps I don't understand C++ well enough but from my understanding, `this` cannot be `nullptr`?

Reply:
`this` cannot be `nullptr`, because if it was, you're invoking a member of `nullptr` which is UB.